src/Microsoft.OData.Core/UriParser/Parsers/UriParserHelper.cs

```diff
@@ -261,6 +261,16 @@ internal static IEdmTypeReference GetLiteralEdmTypeReference(ExpressionTokenKind
             return null;
         }
 
+        /// <summary>
+        /// Determines whether or not an identifier is an annotation term name
+        /// </summary>
+        /// <param name="identifier">The identifer that may be an annotation term name</param>
+        /// <returns>True if the identifier is an annotation term, otherwise false</returns>
+        internal static bool IsAnnotation(string identifier)
+        {
+            return !string.IsNullOrEmpty(identifier) && identifier[0] == '@' && identifier.Contains(".");

```

Comment:
>'@' [](start = 73, length = 3)

Maybe use the existing constants defined in "JsonLightConstants"

Reply:
I'd rather not have the parser reference the jsonlight constants, but I can define '@' in someplace like ODataConstants (in fact, I had done that previously but shelved it as part of the orderby work).

---
In reply to: [166082602](https://github.com/OData/odata.net/pull/1063#discussion_r166082602) [](ancestors = 166082602)

Reply:
Defined a constant in UriQueryConstants for the annotation prefix.

---
In reply to: [166082602](https://github.com/OData/odata.net/pull/1063#discussion_r166082602) [](ancestors = 166082602)